doc/sphinx/practical-tools/utilities.rst

```diff
@@ -452,11 +459,75 @@ To build, say, two targets foo.vo and bar.vo in parallel one can use
 
 
 
-Module dependencies
---------------------
+Building a |Coq| project with Dune
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+
+   the canonical documentation for the Coq Dune extension is
+   maintained upstream; please refer to the `Dune manual
+   <https://dune.readthedocs.io/>`_ for updated information.
+
+Building a Coq project with Dune requires setting up a Dune project
+for your files. This involves adding a ``dune-project`` and
+``pkg.opam`` file to the root (``pkg.opam`` can be empty), and then
+providing ``dune`` files in the directories your ``.v`` files are
+placed. For the experimental version "0.1" of the Coq Dune language,
+|Coq| library stanzas look like:
+
+.. code:: scheme
+
+    (coqlib
+     (name <module_prefix>)
+     (public_name <package.lib_name>)
+     (synopsis <text>)
+     (modules <ordered_set_lang>)
+     (libraries <ocaml_libraries>)
+     (flags <coq_flags>))
+
+This stanza will build all `.v` files in the given directory, wrapping
+the library under ``<module_prefix>``. If you declare a
+``<package.lib_name>`` a ``.install`` file for the library will be
+generated; the optional ``<modules>`` field allows you to filter
+the list of modules, and ``<libraries>`` allows to depend on ML
+plugins. For the moment, Dune does rely on Coq's standard mechanisms
+(such as ``COQPATH`` to locatee installed Coq libraries.
+
+By default Dune will skip ``.v`` files present in subdirectories. In
+order to enable the usual recursive organization of Coq projects add
+
+.. code:: scheme
+
+    (include_subdirs qualified)
 
-In order to compute module dependencies (so to use ``make``), |Coq| comes
-with an appropriate tool, ``coqdep``.
+to you ``dune`` file.
+
+Once your project is set up, `dune build` will generate the
+`pkg.install` files and all the files necessary for the installation
+of your project.
+
+.. example::
+
+   A typical plugin stanza does look like:
+
+.. code:: scheme
+
+    (coqlib
+     (name Equations) ; -R flag
+     (public_name equations.Equations)
+     (synopsis "Equations Plugin")
+     (libraries coq.plugins.extraction equations.plugin)
```

Comment:
Since this is an example of a plugin, it would be nice to also show the dune file of the ML part.

Reply:
Thanks, done.

Reply:
I didn't manage to have example interact well with the code tho.

Reply:
What do you mean?

Reply:
I mean the formatting seems not optimal, likely a silly thing. For example the line introducing the second stanza got absorbed by the code.

Reply:
Fixed it, the example looks good now; thanks @Zimmi48 for all the feedback!